charts/sourcegraph/values.yaml

```diff
@@ -564,6 +558,19 @@ minio:
 pgsql:
   # -- Enable `pgsql` PostgreSQL server
   enabled: true
```

Comment:
@michaellzc - After giving this abstraction some more thought, I actually think the implementation as-is will be a sufficient improvement that we should keep it (and the related minio abstraction).

I'm happy enough with the credentials being stored in their current location, but I'm also ok if we want to move them to a top-level setup, e.g.:

`values.yaml`:
```
credentials:
  pgsql:
    existingSecret:
    database: "sg"
    ...
  minio:
    user: "asdfadsf"
    ...
```

If we keep the credentials alongside the deployments, we could also consider renaming the `pgsql.enabled` flag to `pgsql.external` to make it more clear what that flag will do. I'm uneasy about this though, seems like `enabled` is more standard.

Arguments against the current abstraction:
- **It's incomplete:** if customers want to use an external object store, or use pgbouncer, they'll need to set some environment variables by hand. I don't see this as a significant limitation - they already need to do this, and adding the abstraction doesn't hinder their ability to do this.
- **We may modify this abstraction later and require migration:** We actually reduce the disruption required by enforcing a standard now, because credentials will be stored in a predictable location / single source of truth. If we launch without any abstraction, the database credentials are repeated several times throughout the values.yaml and migration will be worse.

I think the abstraction helps more than it hurts, even if we end up changing something later. It simplifies the process of customizing database/minio credentials, promotes a better practice (using Secrets), and gives us a standard set of keys that we can reference for future migrations.

Reply:
I am sold. let's go with the abstraction.